### DISCUSSION TITLE
Serial Console Barfing Out Garbage

### DISCUSSION THREAD

**Discussion by craigfoo** [SpenceKonde/megaTinyCore#505] [Q&A]
EDIT: Annnd I'm dumb, I didn't "burn bootloader" when I had to switch to 16Mhz. Carry on.

Working with a [ATTiny3216 dev board](https://www.tindie.com/products/drazzy/attiny32161606-dev-board-arduino-compatible/) and I have a FTDI Friend connected in order to spit out serial messages. My baud rates match in code and on the serial console but it still barfs out garbage. What am I missing?

I'm developing with PlatformIO is Visual Studio Code.

Simple Test Code
```
#include <Arduino.h>
// the setup function runs once when you press reset or power the board
void setup() {
  Serial.begin(9600);
  // initialize digital pin LED_BUILTIN as an output.
  pinMode(3, OUTPUT);
}

// the loop function runs over and over again forever
void loop() {
  Serial.print("FUCK");
  digitalWrite(3, HIGH);   // turn the LED on (HIGH is the voltage level)
  delay(1000);                       // wait for a second
  digitalWrite(3, LOW);    // turn the LED off by making the voltage LOW
  delay(1000);                       // wait for a second
}
```
Serial Output
`B˨▒b˨▒b▒▒B˨▒B▒▒▒b˨▒B▒▒▒b˨▒b▒▒▒b▒▒▒B▒▒▒B▒▒▒b▒▒▒B˨▒B▒▒▒B˨▒B▒▒▒B˨▒b˨▒b˨▒B▒▒▒b▒▒▒B▒▒▒b▒▒b▒▒▒`

I've successfully worked with ATTiny816s before, no issue. I wouldn't suspect something to be different with the ATTiny3216.

**Comment by SpenceKonde** [comment]
I don't know if PIO listens to platform.txt sufficiently to do this - but megaTinyCore has automatically set safe fuses (like OSCCFG - any fuses that can't render a system unprogrammable) when uploading through UPDI for a while now. Prior to that, it was the number 1 top reported issue, and may have beeen more commonly reported than all other issues combined!

(the unsafe fuses are the syscfg one that can turn off  UPDI, and the BOD control fuse (since it must be above the BOD voltage to program, but you could set BOD threshold to 4.3v on a chip soldered to the same board as stuff that must run at lower voltages) here on megaTinyCore, and on DxCore, DA series only avoids BODCFG, while the DB series and future DD series treat the syscfg that controls MVIO as unsafe (what happens when you have a dual rail supply connected, and then turn off MVIO? Pretty sure nothing good.... I should probably see how badly that's handled, actually, since I've got real AVR128DB64 boards, and the hacked up ones I did inititial development aren;'t needed anymore. And I even have some of the DIP ones on hand that I could just unplug and throw out, and the only downside risk is that if I burned out the chip, and it missed the trashcan, landed with pins up, and I then stepped on it at night (never done that with a DIP, but I have with pinheader.)

. If there's no configuration option in the menus for something., we don't set the fuse except when doing burn bootloader, on the grounds that if it's anything other than the default, the user set it that way and we shouldn't undo that. Burn bootloader, of course, sets all fuses, so you can always count on the state after burn bootloader being the same.)
